lib/Input.js

```diff
@@ -59,17 +54,21 @@ class Input extends React.Component {
 
     if (this.props.autoresize) {
       // scrollWidth is designed to be fast not accurate.
-      // +2 is completely arbitrary but does the job.
-      inputWidth = Math.ceil(this.sizer.scrollWidth) + 2
+      // changed +2 to +20 to get the same behavior
+      inputWidth = Math.ceil(this.getScrollWidth()) + 20
+      return inputWidth
     }
+  }
 
-    if (inputWidth !== this.state.inputWidth) {
-      this.setState({ inputWidth })
+  getScrollWidth () {
+    if (this.sizer) {
+      return this.sizer.scrollWidth       // we have to mock this function in test to return a random number
     }
   }
 
   render () {
     const { inputAttributes, inputEventHandlers, query, placeholder, expandable, listboxId, selectedIndex } = this.props
+    const width = this.updateInputWidth()
```

Comment:
I don't think this will get the correct width as the latest text may not have been rendered yet. I assume this is why the extra width (2 => 20) was required but I don't think this can be relied upon 🤔

Reply:
After testing code I can confirm this is always one step behind. I'll look into the error you describe but I haven't yet been able to recreate it myself.

![step-behind](https://user-images.githubusercontent.com/271645/52968167-524de300-33a4-11e9-80ed-744750427f7d.gif)